clients/python/girder_client/__init__.py

```diff
@@ -486,6 +486,60 @@ def uploadFileToItem(self, itemId, filepath, reference=None):
                                 ' not receive object with _id. Got instead: ' +
                                 json.dumps(obj))
 
+    """
+        Uploads contents of a file
+
+        :param uploadObj: The upload object contain the upload id.
+        :type uploadObj: dict
+        :param stream: Readable stream object.
+        :type stream: file-like
+        :param size: The length of the file. This must be exactly equal to the
+            total number of bytes that will be read from ``stream``, otherwise
+            the upload will fail.
+        :type size: str
+        :param progressCallback: If passed, will be called after each chunk
+            with progress information. It passes a single positional argument
+            to the callable which is a dict of information about progress.
+        :type progressCallback: callable
+    """
+    def _uploadContents(self, uploadObj, stream, size, progressCallback=None):
+        offset = 0
+        uploadId = uploadObj['_id']
+        while True:
+            data = stream.read(self.MAX_CHUNK_SIZE)
```

Comment:
I am happy to make the change you suggest, however, in that case the upload would fail as you are try to upload more data than was specified I think.